### ISSUE TITLE
verify_install_requires no longer checks for pytest runtime dependencies

### ISSUE THREAD

**Issue by jdavies-st** [spacetelescope/jwst#8379]
The original purpose of `verify_install_requires` was to prevent there being runtime imports that are not specified in `install_requires`, but specifically ones that are not caught because some runtime import was being fullfilled by a test dependency, specifically `pytest` pulling in a dependency needed at runtime but not listed in `install_requires`.  This had happened repeatedly - we didn't see a problem in CI testing because `[test]` extras pulled in the requirement, but then when users in stalled the package without `[test]` extras, it failed.  See https://github.com/spacetelescope/jwst/pull/4979

Currently the new version of `verify_install_requires` specifically depends on `pytest`, which short-circuits this check.

https://github.com/spacetelescope/jwst/blob/42f0e01ad64f7e275a811c3a439fa75927bd3690/jwst/scripts/verify_install_requires.py#L26-L32

and

https://github.com/spacetelescope/jwst/blob/42f0e01ad64f7e275a811c3a439fa75927bd3690/jwst/tests/test_import.py#L6-L29

This should be refactored to exclude pytest, and to make sure when it is called by tox, that no `[test]` extras are installed.

**Comment by braingram** [comment]
The script is run in an environment where `[test]` extras are not installed:
https://github.com/spacetelescope/jwst/blob/64a92a962300b32039e69376ccc399a3c80f1634/tox.ini#L16-L23

**Comment by jdavies-st** [comment]
Yes, but `pytest` and all its dependencies are pulled in before the dependency check is done.  `pytest` and its dependencies are not pulled in when I do a `pip install jwst`.  So it is not checking what you think it is checking.  This was precisely the case for creating `verify_install_requires` in the first place.

**Comment by braingram** [comment]
Are you concerned about pytest being a silent dependency?

**Comment by jdavies-st** [comment]
Yes, both `pytest` and its dependencies, both of which have been issues in the past, which was the main reason for creating this dependency check.

https://github.com/pytest-dev/pytest/blob/2b40981021886afd55b0cd18cb549bf5c44bf120/pyproject.toml#L48-L55

**Comment by braingram** [comment]
Thanks. It sounds like from the original concerns:
- `make sure when it is called by tox, that no [test] extras are installed`
- `refactored to exclude pytest`

The first one is currently addressed.
